### PR TITLE
Enforce `gofmt` in GitHub Actions

### PR DESCRIPTION
**Pull request by tianon**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -36,5 +36,7 @@ jobs:
           name: coverage
           path: .test/coverage**
           if-no-files-found: error
+      - name: gofmt
+        run: find -name '*.go' -type f -exec ./.go-env.sh gofmt -l -s -w '{}' +
```
Comment (whalelines) [inline .github/workflows/ci.yml:40]: Do you not need to supply a path to `find`?

**Review by whalelines — COMMENTED** [review]
It would be nice if `gofmt` returned non-zero when using `-l` or `-d`.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tianon) [reply .github/workflows/ci.yml:40]: It defaults to the current working directory :+1:

**Comment by tianon** [comment]
> It would be nice if `gofmt` returned non-zero when using `-l` or `-d`.

I think it _can_ be convinced to do so, but this way makes sure that our following `git diff` invocation catches any/all of them _and_ any other missed things from our "canonical JSON" (instead of fixing one error only to find another), so I think this might actually be slightly better for our case. :smile:

**Review by whalelines — APPROVED** [review]
LGTM

File: .github/workflows/ci.yml (hunk 1 above)
Comment (whalelines) [reply .github/workflows/ci.yml:40]: Must be a GNU find enhancement. BSD find errors.

Comment (tianon) [reply .github/workflows/ci.yml:40]: Even BusyBox is happy with it! :sweat_smile: 

```console
$ docker run --rm busybox find -maxdepth 1 -name etc
./etc
```

Comment (whalelines) [reply .github/workflows/ci.yml:40]: Should I read that as a plus or a minus?
/me ducks

**Review by yosifkit — APPROVED** [review]

**Pull request merged**